sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/rust/RustGenerator.java

```diff
@@ -790,7 +790,7 @@ private static void generatePrimitiveRequiredDecoder(
 
         if (fieldToken.version() > 0)
         {
-            indent(sb, level + 1, "if self.acting_version < %d {\n", fieldToken.version());
+            indent(sb, level + 1, "if self.acting_version > 0 && self.acting_version < %d {\n", fieldToken.version());
```

Comment:
I wonder why `self.acting_version > 0` condition is added.
Is `self.acting_version < %d` is not enough for version checking?

Reply:
I figured out why `self.acting_version > 0` is required.
acting_version is set to 0 when there is no acting_version requirement.